packages/flutter/lib/src/material/outline_button.dart

```diff
@@ -370,14 +379,26 @@ class _OutlineButtonState extends State<_OutlineButton> with SingleTickerProvide
     return colorTween.evaluate(_fillAnimation);
   }
 
+  Color get _currentOutlineColor {
+    if (!widget.enabled)
+      return widget.disabledBorderColor;
+    if (_pressed)
+      return widget.highlightedBorderColor;
+    return widget.borderSide?.color;
+  }
+
   BorderSide _getOutline() {
     if (widget.borderSide?.style == BorderStyle.none)
       return widget.borderSide;
 
-    final Color specifiedColor = widget.enabled
-      ? (_pressed ? widget.highlightedBorderColor : null) ?? widget.borderSide?.color
-      : widget.disabledBorderColor;
-
+    Color specifiedColor;
+    // If outline color is a `MaterialStateProperty`, it will be used in all
+    // states, otherwise we determine the outline color in the current state.
+    if (widget.borderSide?.color is MaterialStateProperty<Color>) {
+      specifiedColor = widget.borderSide?.color;
+    } else {
+      specifiedColor = _currentOutlineColor;
```

Comment:
I think it may be better to not use a getter here since that getter is only used in a single place and relies on borderSide.color not being a MaterialStateProperty. 

Alternatively (and preferably), you could update the getter to include the additional check for MaterialStateProperty

Reply:
Good call, moved the `MaterialStateProperty` check to the getter.